web/modules/custom/registration_form_custom/src/Form/NewUserRegisterForm.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace Drupal\registration_form_custom\Form;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Entity\EntityRepositoryInterface;
+use Drupal\Core\Entity\EntityTypeBundleInfoInterface;
+use Drupal\Core\Entity\EntityTypeManager;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\Locale\CountryManager;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\user\Entity\User;
+use Drupal\user\RegisterForm;
+use Drupal\weather_block\Services\FetchApiData;
+use Drupal\weather_block\Services\UserWeatherHandler;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a user register form.
+ */
+class NewUserRegisterForm extends RegisterForm {
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct(EntityRepositoryInterface $entity_manager,
+    EntityRepositoryInterface $entity_repository,
+    LanguageManagerInterface $language_manager,
+    EntityTypeBundleInfoInterface $entityTypeBundleInfo = NULL,
+    TimeInterface $time = NULL,
+    ModuleHandlerInterface $moduleHandler,
+    public UserWeatherHandler $cityService,
+    private CountryManager $countryManager,
+    private EntityTypeManager $typeManager,
+    protected AccountProxyInterface $accountProxy,
+    protected FetchApiData $apiData,
+    protected ConfigFactoryInterface $configFact,
+    protected Connection $database,
+  ) {
+    $user = User::create();
+    $this->setEntity($user);
+    $this->setModuleHandler($moduleHandler);
+
+    parent::__construct($entity_repository, $language_manager, $entityTypeBundleInfo, $time);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity.repository'),
+      $container->get('entity.repository'),
+      $container->get('language_manager'),
+      $container->get('entity_type.bundle.info'),
+      $container->get('datetime.time'),
+      $container->get('module_handler'),
+      $container->get('weather_block.get_set_city_user'),
+      $container->get('country_manager'),
+      $container->get('entity_type.manager'),
+      $container->get('current_user'),
+      $container->get('weather_block.fetch_api_data'),
+      $container->get('config.factory'),
+      $container->get('database'),
+    );
+  }
+
+  /**
+   * Return form ID.
+   */
+  public function getFormId() {
+    return 'custom_registration_form';
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function form(array $form, FormStateInterface $form_state): array {
+
+    $form = parent::form($form, $form_state);
+    $cities = $this->cityService->getCitiesArray();
+    $term_options = $this->getTaxonomyTerm();
+
+    $form['password'] = [
```

Comment:
isn't password declared in parent form?

Reply:
![image](https://github.com/HeorhiiBoliuk/Drupal10Bootcamp/assets/109587214/1aadb914-66a6-46d8-bd09-1af244f897ba)
/admin/config/people/accounts